zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbProcessState.java

```diff
@@ -159,6 +185,17 @@ public void putProcess(final long key, final ProcessRecord processRecord) {
     putLatestVersionDigest(processRecord);
   }
 
+  @Override
+  public void storeProcessDefinitionKeyByProcessIdAndDeploymentKey(
+      final ProcessRecord processRecord) {
+    tenantIdKey.wrapString(processRecord.getTenantId());
+    processDefinitionKey.wrapLong(processRecord.getProcessDefinitionKey());
+    processId.wrapBuffer(processRecord.getBpmnProcessIdBuffer());
+    deploymentKey.wrapLong(processRecord.getDeploymentKey());
+    processDefinitionKeyByProcessIdAndDeploymentKeyColumnFamily.upsert(
+        tenantAwareProcessIdAndDeploymentKey, fkProcessDefinitionKey);
```

Comment:
I used `insert` first, but the following test was failing then: `MultiPartitionDeploymentLifecycleTest.shouldRejectCompleteDeploymentDistributionWhenAlreadyCompleted`
It looked like a `ZeebeDbInconsistentException` prevented further records (that are expected in the test) from being written.
I didn't feel very comfortable with re-writing that test and wasn't sure either if there might actually be situations where a previous entry would be overwritten, so I switched to `upsert`.

Reply:
Oh that's interesting and although not obvious, it makes sense.

Deployment is distributed among partitions. This applies a retry mechanism, and it can happen that the distributed `Deployment:Create` command is send to (and processed) multiple times on the receiving partition. So, this command must be processed idempotently. It doesn't do any look ups of the deployed resources and thus can overwrite the ones created in previous distribution attempts.

Reply:
Thanks for the explanation!